gse-copy-paste-afs/src/main/java/com/powsybl/gse/copy_paste/afs/CopyManager.java

```diff
@@ -0,0 +1,269 @@
+        projectFolder.unarchive(info.archivePath.resolve(info.nodeId));
+        projectFolder.getChild(name).ifPresent(pNode -> {
+            if (!name.contains(copyDuplicated) && !projectFolder.getChild(name + copyDuplicated).isPresent()) {
+                pNode.rename(name + copyDuplicated);
+                child.rename(name);
+            } else {
+                if (!name.contains(copyDuplicated)) {
+                    renameCopiedNode(projectFolder, copyDuplicated, child, name, pNode);
+                } else {
+                    renameCopiedNode(projectFolder, "", child, name, pNode);
+                }
+            }
+        });
+    }
+
+    private static void renameCopiedNode(ProjectFolder projectFolder, String copy, AbstractNodeBase child, String name, ProjectNode projectNode) {
+        String lastCopiedNode = projectFolder.getChildren().stream()
+                .map(ProjectNode::getName)
+                .filter(n -> n.startsWith(name + copy))
+                .sorted(Comparator.reverseOrder())
+                .collect(Collectors.toList())
+                .get(0);
+
+        String lastCopiedNodeNumber = lastCopiedNode.substring(lastCopiedNode.indexOf(name + copy) + name.length() + copy.length());
+        if (lastCopiedNodeNumber.isEmpty()) {
+            projectNode.rename(name + copy + 2);
+            child.rename(name);
+        } else {
+            int numberOfCopies = Integer.parseInt(lastCopiedNodeNumber) + 1;
+            projectNode.rename(name + copy + numberOfCopies);
+            child.rename(name);
+        }
+    }
+
+    private void archiveAndCopy(AbstractNodeBase node) {
+        String nodeId = node.getId();
+        String archiveDirectory = storageDirectory + node.getName() + node.getId();
+        archiveNode(node, nodeId, archiveDirectory);
+    }
+
+    private static void archiveNode(AbstractNodeBase node, String nodeId, String parentPath) {
+        java.io.File archiveRootFolder = new java.io.File(parentPath);
+        if (!archiveRootFolder.exists()) {
+            archiveRootFolder.mkdir();
+        }
+        java.io.File archiveDestinationFolder = new File(parentPath, nodeId);
+        if (archiveDestinationFolder.exists()) {
+            try {
+                FileUtils.deleteDirectory(archiveDestinationFolder);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        node.archive(Paths.get(parentPath));
+        deleteOnExit(archiveRootFolder);
+    }
+
+    private static void deleteOnExit(File folder) {
+        folder.deleteOnExit();
+        for (File file : folder.listFiles()) {
+            if (file.isDirectory()) {
+                deleteOnExit(file);
+            } else {
+                file.deleteOnExit();
+            }
+        }
+    }
+
+    private static String getStorageDirectory() {
+        return LocalComputationConfig.load().getLocalDir() + CopyServiceConstants.PATH_SEPARATOR;
```

Comment:
Why use CopyServiceConstants.PATH_SEPARATOR and not the more explicit constant (proxyied by this constant.) `FileSystems.getDefault().getSeparator()` ?